sklearn/utils/class_weight.py

```diff
@@ -69,7 +72,12 @@ def compute_class_weight(class_weight, *, classes, y):
         if not all(np.isin(classes, le.classes_)):
             raise ValueError("classes should have valid labels that are in y")
 
-        recip_freq = len(y) / (len(le.classes_) * np.bincount(y_ind).astype(np.float64))
+        n_effective_samples = np.bincount(y_ind, weights=sample_weight).sum()
```

Comment:
The result `np.bincount(y_ind, weights=sample_weight)` should be stored in a local variable to avoid computing it twice in a row.